### PR TITLE
Discord: Show game name and a core icon

### PR DESCRIPTION
**Pull request by fr500**
image

Guess I'll do it myself.
So here is what needs to be done:

LibRetro should create it's own discord app and change the ID, I'm using a testing APP **I own** for this and I'll nuke it, change icons to dicks or whatever I want any time I want to.

Creating the app is straightforward anyway, just add an icon, fill the app name:

image

Then add the assets:

image

For this implementation I'm using core_name (lowercase, spaces removed), so something would have to upload icons with the corenames (again lowercase, spaces removed) to discord. 

I tried just using system_name but it seems the discord API doesn't like long strings and there is nothing in the documentation that indicates a max size (https://discordapp.com/developers/docs/rich-presence/how-to) So I wouldn't know how to truncate them otherwise (so core name seemed easier)

Basically the assets uploaded have to have the same name as the return value of this:
```C
const char *system_name  = string_replace_substring(string_to_lower(core_info->core_name), " ", "_");
```

That's all!

### CODE REVIEW

**Comment by fr500** [comment]
canvas2

Here is the PNG I used for the icon (and "base" icon)

super nintendo entertainment system

And the snes9x icon
